app/components/Image/Image.tsx

```diff
@@ -37,20 +45,28 @@ const ImageComponent = (props: Props) => {
       setImageError(false);
       setLoadStart(Date.now());
     }
-
+    window.addEventListener('themeChange', () => {
+      setThemedSource(
+        getTheme() && props.darkThemeSource
+          ? (getTheme() ?? 'light') === 'light'
+            ? src
+            : props.darkThemeSource
+          : src
+      );
+    });
```

Comment:
```suggestion
    const onThemeChange = () => {
      setThemedSource(
        getTheme() && props.darkThemeSource
          ? (getTheme() ?? 'light') === 'light'
            ? src
            : props.darkThemeSource
          : src
      );
    window.addEventListener('themeChange', onThemeChange});
	return () => window.removeEventListener('themeChange', onThemeChange);
```
	
Remember to unregister the event listener i the cleanup (returned) function. If not a new event listener will be added on each render but never removed, causing the app to slow down after a while.

Reply:
Oh, and also add an empty dependency array to avoid running the effect unnecessarily. No dep-array means it runs on every rerender.

Reply:
This should also be in a separate useEffect with `[src, props.darkThemeSrc]` as dependencies, to make sure it works if src changes after initializing the component.